src/steamship/agents/service/agent_service.py

```diff
@@ -295,6 +306,33 @@ def build_default_context(self, context_id: Optional[str] = None, **kwargs) -> A
         context = with_llm(context=context, llm=llm)
         return context
 
+    @post("async_prompt")
+    def async_prompt(
+        self, prompt: Optional[str] = None, context_id: Optional[str] = None, **kwargs
+    ) -> StreamingResponse:
```

Comment:
should we just use `InvocableResponse(status=task, data=ctx.chat_history.file)` instead?

Reply:
Data = the file makes sense to me, rather than having a new type.  See below, I'm not sure I yet understand the task bit.

Reply:
It's also possible the caller might want a more direct reference to the new blocks that are being streamed as a result of this request?

Reply:
That feels like a really clean return type to me

Reply:
@dkolas we don't have any way to cascade pre-allocated blocks **within** a `File` back -- and I'm not entirely sure that we can (given the current `Agent` + `AgentService` architecture). For instance, the status message blocks, by their very nature, cannot be pre-allocated. Any scheme of pre-allocation would then result in out-of-order messaging, etc.

I _think_ `File` may be the "best" we can do easily. For things that don't require an `Agent` and `Action` and status messages, that may not be the case.

Let's talk this through in person.